### PR TITLE
Remove Karma libraries

### PR DESCRIPTION
**Pull request by antkiewiczk**
Scope of changes:

1. examples/website:
- remove @karma.run/utility library and replace utils functions inside project
2. packages/api:
- remove @karma.run/utility library and replace utils functions inside project
- fix some types
3. packages/editor
- remove @karma.run/utility library and replace utils functions inside project
- remove @wepublish/karma.run-react library
- remove @karma.run/react library
- replace routing throughout the application with react-router
- IMPORTANT: change nested routing pattern to follow the industry standard -> nested route extends the path of parent route (i.e. articles -> articles/create instead of  articles -> article/create)
- introduce eslint sorting plugin to improve code readability and tidiness

### CODE REVIEW

File: packages/editor/src/client/atoms/user/userSubscriptionsList.tsx [hunk 1]
```diff
@@ -123,11 +124,11 @@ export function UserSubscriptionsList({subscriptions}: UserSubscriptionsProps) {
             </FlexboxGrid.Item>
             {/* edit subscription */}
             <FlexboxGrid.Item colspan={6} style={{textAlign: 'right'}}>
-              <ButtonLink
-                appearance="ghost"
-                route={SubscriptionEditRoute.create({id: subscription.id})}>
-                <EditIcon /> {t('userSubscriptionList.editSubscription')}
-              </ButtonLink>
+              <Link to={`/subscriptions/edit/${subscription.id}`}>
```
Comment (penk85) [inline packages/editor/src/client/atoms/user/userSubscriptionsList.tsx:127]: Why did you change this?

File: packages/editor/src/client/route.tsx [hunk 2]
```diff
@@ -1,308 +0,0 @@
-import React, {ReactNode, useContext} from 'react'
```
Comment (penk85) [inline packages/editor/src/client/route.tsx:1]: Why did you delete this?

File: packages/editor/src/client/routes/articleEditor.tsx [hunk 3]
```diff
@@ -563,8 +546,23 @@ export function ArticleEditor({id}: ArticleEditorProps) {
...
+                    size={'lg'}
+                    icon={<EyeIcon />}
+                    // onClick={e => {
+                    //   previewLinkFetch({
+                    //     variables: {
+                    //       id: id!,
+                    //       hours: 1
+                    //     }
+                    //   })
+                    // }}
```
Comment (penk85) [inline packages/editor/src/client/routes/articleEditor.tsx:561]: Why did you remove this?

File: packages/editor/src/client/routes/pageList.tsx [hunk 4]
```diff
@@ -432,10 +428,7 @@ export function PageList() {
                     }
                   }).then(output => {
                     if (output.data) {
-                      dispatch({
-                        type: RouteActionType.ReplaceRoute,
-                        route: PageEditRoute.create({id: output.data?.duplicatePage.id})
-                      })
+                      navigate(`/pages/edit/${output.data?.duplicatePage.id}`, {replace: true})
```
Comment (penk85) [inline packages/editor/src/client/routes/pageList.tsx:431]: Why this change?

File: packages/api/src/utility.ts [hunk 5]
```diff
@@ -249,6 +249,30 @@ export function countRichtextChars(blocksCharLength: number, nodes: any) {
...
+
+export type MapDiscriminatedUnion<T extends Record<K, string>, K extends keyof T> = {
+  [V in T[K]]: DiscriminateUnion<T, K, V>
+}
+
+export function isObject<T>(unknown: unknown): unknown is Record<any, T> {
+  return typeof unknown === 'object' && unknown != null && !Array.isArray(unknown)
+}
+
+export function isArray<T>(unknown: unknown): unknown is T[] {
```
Comment (Itrulia) [inline packages/api/src/utility.ts:264]: love the usage of `is` <3

**Review by Itrulia — APPROVED** [review]
Absolutely love it!

**Review by penk85 — DISMISSED** [review]

**Review by Itrulia — DISMISSED** [review]

**Review by Itrulia — APPROVED** [review]

**Review by penk85 — APPROVED** [review]

**Review by tomaszdurka — APPROVED** [review]
